corehq/motech/views.py

```diff
@@ -180,3 +180,46 @@ def post(self, *args, **kwargs):
         return self.paginate_crud_response
 
 
+@method_decorator(require_permission(Permissions.edit_motech), name='dispatch')
+class ConnectionSettingsDetailView(BaseProjectSettingsView, ModelFormMixin, ProcessFormView):
+    urlname = 'connection_settings_detail_view'
+    page_title = _('Connection Settings')
+    template_name = 'motech/connection_settings_detail.html'
+    model = ConnectionSettings
+    form_class = ConnectionSettingsForm
+
+    def dispatch(self, request, *args, **kwargs):
+        # TODO: When Repeaters use Connection Settings, drop, and use
+        # @requires_privilege_with_fallback(privileges.DATA_FORWARDING)
+        if not (
+                toggles.DHIS2_INTEGRATION.enabled_for_request(request)
+                or toggles.INCREMENTAL_EXPORTS.enabled_for_request(request)
+        ):
+            raise Http404()
+        return super().dispatch(request, *args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        # Allow us to update if 'pk' is given in the URL, otherwise create
+        self.object = self.get_object() if self.pk_url_kwarg in self.kwargs else None
```

Comment:
Does this stop you somehow from being able to edit other people's connection settings?

Reply:
Urgh! Thank you! [afae297](https://github.com/dimagi/commcare-hq/pull/27527/commits/afae2976f1fc4c1a7f3297a07b1951934604d70b)

Reply:
:stuck_out_tongue_winking_eye: